arangodb-net-standard/CursorApi/PostCursorOptions.cs

```diff
@@ -1,11 +1,108 @@
+        /// It can be used to count the number of documents that match
+        /// certain filter criteria, but only return a subset of them, in one go.
+        /// It is thus similar to MySQL’s SQL_CALC_FOUND_ROWS hint.
+        /// Note that setting the option will disable a few LIMIT optimizations
+        /// and may lead to more documents being processed, and thus make queries run longer.
+        /// Note that the fullCount attribute may only be present in the result if the query
+        /// has a top-level LIMIT clause and the LIMIT clause is actually used in the query.
+        /// </summary>
+        public bool? FullCount { get; set; }
+
+        /// <summary>
+        /// Limits the maximum number of plans that are created by the AQL query optimizer.
+        /// </summary>
+        public long? MaxPlans { get; set; }
+
+        /// <summary>
+        /// Limits the maximum number of warnings a query will return.
+        /// The number of warnings a query will return is limited to 10 by default,
+        /// but that number can be increased or decreased by setting this attribute.
+        /// </summary>
+        public long? MaxWarningCount { get; set; }
+
+        /// <summary>
+        /// When set to true, the query will throw an exception and abort
+        /// instead of producing a warning. This option should be used during development
+        /// to catch potential issues early. When the attribute is set to false,
+        /// warnings will not be propagated to exceptions and will be returned with the query result.
+        /// There is also a server configuration option --query.fail-on-warning for setting
+        /// the default value for failOnWarning so it does not need to be set on a per-query level.
+        /// </summary>
+        public bool? FailOnWarning { get; set; }
+
+        /// <summary>
+        /// Specify true and the query will be executed in a streaming fashion. The query result
+        /// is not stored on the server, but calculated on the fly. Beware: long-running queries
+        /// will need to hold the collection locks for as long as the query cursor exists.
+        /// When set to false a query will be executed right away in its entirety. In that case
+        /// query results are either returned right away (if the result set is small enough),
+        /// or stored on the arangod instance and accessible via the cursor API (with respect to the ttl).
+        /// It is advisable to only use this option on short-running queries or without exclusive locks
+        /// (write-locks on MMFiles). Please note that the query options cache,
+        /// count and fullCount will not work on streaming queries. Additionally query statistics,
+        /// warnings and profiling data will only be available after the query is finished.
+        /// The default value is false
+        /// </summary>
+        public bool? Stream { get; set; }
+
+        // optimizer.rules ?
```

Comment:
According to the example in the docs, a list of rules can be provided to the optimizer:

```json
{ 
  "query" : "FOR i IN 1..10 LET a = 1 LET b = 2 FILTER a + b == 3 RETURN i", 
  "count" : true, 
  "options" : { 
    "maxPlans" : 1, 
    "optimizer" : { 
      "rules" : [ 
        "-all", 
        "+remove-unnecessary-filters" 
      ] 
    } 
  } 
}
```

Reply:
Thanks, guess I didn't find that.